designer/server/src/main/scala/pl/touk/nussknacker/ui/config/DesignerConfig.scala

```diff
@@ -1,62 +1,242 @@
+    val surveySettings: Option[SurveySettings],
+    val tabs: Option[List[TopTab]],
+    val intervalTimeSettings: IntervalTimeSettings,
+    val testDataSettings: TestDataSettings,
+    val enableConfigEndpoint: Boolean,
+    val redirectAfterArchive: Boolean,
+    val componentDefinitionExtractionMode: ComponentDefinitionExtractionMode,
+    val stickyNotesSettings: StickyNotesSettings,
+    val deploymentsStatusesSynchronizationConfig: DeploymentsStatusesSynchronizationConfig,
+    val finishedDeploymentStatusesSynchronization: FinishedDeploymentsStatusesSynchronizationConfig,
+    val componentLinks: List[ComponentLinkConfig],
+    val processToolbarConfig: CategoriesScenarioToolbarsConfig,
+    val questDbSettings: QuestDbConfig,
+    val notifications: NotificationConfig,
+    val fragmentPropertiesDocsUrl: Option[String],
+    val repositoryGaugesCacheDuration: Duration,
+    val globalBuildInfo: Option[Map[String, String]],
+    val ssl: Option[KeyStoreConfig],
+    val http: HttpConfig,
+    val attachments: AttachmentsConfig,
+) {
 
-  import DesignerConfig._
+  // TODO: We should parse configuration options to fields instead of accessing rawConfig. Thank to that:
+  //       - we will get errors faster if there is some problem in configuration
+  //       - structure of the config will be visible in classes
+  val rawConfig: Config = rawConfigWithUnresolvedVersion.resolved
 
-  def processingTypeConfigs: ProcessingTypeConfigs =
-    ProcessingTypeConfigs(processingTypeConfigsRaw.asMap.mapValuesNow(ProcessingTypeConfig.read))
+  def processingTypeConfigs(): ProcessingTypeConfigs =
+    ProcessingTypeConfigs(processingTypeConfigsRaw().asMap.mapValuesNow(ProcessingTypeConfig.read))
 
-  def processingTypeConfigsRaw: ConfigWithUnresolvedVersion =
-    rawConfig
+  def processingTypeConfigsRaw(): ConfigWithUnresolvedVersion =
+    rawConfigWithUnresolvedVersion
       .getConfigOpt("scenarioTypes")
       .getOrElse {
         throw ConfigurationMalformedException("No scenario types configuration provided")
       }
 
-  def configLoaderConfig: Config = rawConfig.resolved.getAs[Config]("configLoader").getOrElse(ConfigFactory.empty())
+  def render(): String = rawConfig.root().render(ConfigRenderOptions.concise())
+
+}
+
+object DesignerConfig {
+
+  private[config] val defaultConfigResource = "defaultDesignerConfig.conf"
+
+  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+  import net.ceedubs.ficus.readers.EnumerationReader._
+  import pl.touk.nussknacker.engine.util.config.CustomFicusInstances._
+
+  def from(config: Config): DesignerConfig = {
+    val defaultConfig = ConfigFactory.parseResources(getClass.getClassLoader, DesignerConfig.defaultConfigResource)
+    val configWithFallbackToDefault = config.withFallback(defaultConfig)
+    DesignerConfig(ConfigWithUnresolvedVersion(configWithFallbackToDefault))
+  }
+
+  def apply(rawConfig: ConfigWithUnresolvedVersion): DesignerConfig = {
```

Comment:
Now we have `from` and `apply` methods. Can we use only one name to have two methods with different, overloaded parameters?